### PR TITLE
Use LEGACYFULL instead of FULL in the documentation.

### PR DESCRIPTION
**Pull request by YohannDudouit**
I had forgotten some references to `AssemblyLevel::FULL` in the documentation, this PR uses the correct default mode `AssemblyLevel::LEGACYFULL`. I also removed some documentation that seemed outdated to me.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1683](https://github.com/mfem/mfem/pull/1683) | @YohannDudouit | @tzanio | @tzanio + @v-dobrev | 08/13/20 | 10/05/20 | 10/08/20 | |

### CODE REVIEW

File: fem/bilinearform.hpp
Comment (tzanio) [inline fem/bilinearform.hpp:160]:
```suggestion
       - AssemblyLevel::LEGACYFULL (default)
```

File: fem/complex_fem.hpp
Comment (tzanio) [inline fem/complex_fem.hpp:225]:
```suggestion
       - AssemblyLevel::LEGACYFULL (default)
```

File: fem/complex_fem.hpp
Comment (tzanio) [inline fem/complex_fem.hpp:546]:
```suggestion
       - AssemblyLevel::LEGACYFULL (default)
```

**Comment by tzanio** [comment]
@acfisher, can you please review?

**Comment by tzanio** [comment]
ping: @acfisher

**Comment by v-dobrev** [comment]
One observation: "full assembly" can refer to two slightly different operator representations: the "full true-dof parallel assembly" and the "full processor-local assembly", see https://ceed.exascaleproject.org/bps/#terminology-and-notation. In the first case (which is what `LEGACYFULL` does) the operator is represented as a `HypreParMatrix` including the action of the parallel prolongation, P. In the second case (which is what `FULL` does now) only the processor-local matrices are assembled as `SparseMatrix` and P is applied on-the-fly.

From this point of view, it may be natural to add another assembly option for "full true-dof parallel assembly", e.g. `FULL_GLOBAL`. This option will be similar to `LEGACYFULL` but it can use `FULL` assembly plus parallel RAP to form a `HypreParMatrix` (or other parallel sparse matrix formats).

**Comment by YohannDudouit** [comment]
@v-dobrev Should I improve the documentation of `FULL` and `LEGACYFULL`?

**Comment by v-dobrev** [comment]
> @v-dobrev Should I improve the documentation of `FULL` and `LEGACYFULL`?

It could be useful.

The main question for me is: what do we want `FULL` to mean? Is it just "full processor-local assembly", i.e. processor-local sparse matrices? Or does it also include the "full true-dof parallel assembly", i.e. parallel sparse matrices? This is a question for everyone. I personally prefer if we have different assembly levels for these two, just to be more explicit.

**Comment by tzanio** [comment]
> The main question for me is: what do we want `FULL` to mean? Is it just "full processor-local assembly", i.e. processor-local sparse matrices? Or does it also include the "full true-dof parallel assembly", i.e. parallel sparse matrices?

My preference will be to use `FULL` for the "full true-dof parallel assembly". If we want to keep it short, we can call the one where we stop at MPI task-local sparse matrices `DD` for domain-decomposed.

**Comment by v-dobrev** [comment]
> My preference will be to use `FULL` for the "full true-dof parallel assembly". If we want to keep it short, we can call the one where we stop at MPI task-local sparse matrices `DD` for domain-decomposed.

One problem with this is that, currently, `FULL` actually performs the other action, "full processor-local assembly".

**Comment by tzanio** [comment]
> One problem with this is that, currently, FULL actually performs the other action, "full processor-local assembly".

Hmmm, that's less useful for H1

**Comment by tzanio** [comment]
We shouldn't hold this PR, as it improves the documentation, but we should take as an action item to implement full FA.

My preference will be to call that "FULL" and come up with a different name (e.g. "LA" for L-vector assembly) for the currently implemented option.

@acfisher, can you approve if you agree?

**Comment by tzanio** [comment]
@v-dobrev if you agree with ☝️ can you approve?

**Comment by v-dobrev** [comment]
> @v-dobrev if you agree with ☝️ can you approve?

I'm not a reviewer, so there's no need to wait for my approval. I was just pointing out an inconsistency which can be resolved later.

**Review by v-dobrev — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**
